--- vienna/FeedCredentials.m	2026-04-05 14:14:13.116202968 +0000
+++ vienna/FeedCredentials.m	2026-04-06 21:48:19.433830125 +0000
@@ -59,7 +59,7 @@
 
 	// Show the feed URL in the prompt so the user knows which site credentials are being
 	// requested. (We don't use [folder name] here as that is likely to be "Untitled Folder" mostly).
-	NSURL * secureURL = [NSURL URLWithString:[folder feedURL]];
+	NSURL * secureURL = [NSURL XP_URLWithString:[folder feedURL]];
 	NSString * prompt = [NSString stringWithFormat:NSLocalizedString(@"Credentials Prompt", nil), [secureURL host]];
 	[promptString setStringValue:prompt];
 
