--- vienna/FoldersTree.m	2026-04-07 13:09:04.212992030 +0000
+++ vienna/FoldersTree.m	2026-04-07 13:11:21.977295725 +0000
@@ -23,7 +23,6 @@
 #import "AppController.h"
 #import "Constants.h"
 #import "Preferences.h"
-#import "HelperFunctions.h"
 #import "StringExtensions.h"
 #import "FolderView.h"
 #import "PopupButton.h"
@@ -243,10 +242,11 @@
  */
 -(void)unarchiveState:(NSArray *)stateArray
 {
-	for (NSDictionary * dict in stateArray)
+	NSEnumerator * enum_dict = [stateArray objectEnumerator];
+	NSDictionary * dict;
+	while ((dict = [enum_dict nextObject]))
 	{
-		int folderId = [[dict valueForKey:@"NodeID"] intValue];
-		TreeNode * node = [rootNode nodeFromID:folderId];
+		int folderId = [[dict valueForKey:@"NodeID"] intValue];		TreeNode * node = [rootNode nodeFromID:folderId];
 		if (node != nil)
 		{
 			BOOL doExpandItem = [[dict valueForKey:@"ExpandedState"] boolValue];
@@ -275,10 +275,10 @@
 	Folder * folder;
 	if ([[Preferences standardPreferences] foldersTreeSortMethod] != MA_FolderSort_Manual)
 	{
-		for (folder in listOfFolders)
+		NSEnumerator * enum_folder = [listOfFolders objectEnumerator];
+		while ((folder = [enum_folder nextObject]))
 		{
-			int itemId = [folder itemId];
-			NSArray * listOfSubFolders = [[Database sharedDatabase] arrayOfFolders:itemId];
+			int itemId = [folder itemId];			NSArray * listOfSubFolders = [[Database sharedDatabase] arrayOfFolders:itemId];
 			int count = [listOfSubFolders count];
 			TreeNode * subNode;
 
@@ -1150,7 +1150,7 @@
 			[stringDragData appendString:feedURL];
 			[stringDragData appendString:@"\n"];
 			
-			NSURL * safariURL = [NSURL URLWithString:feedURL];
+			NSURL * safariURL = [NSURL XP_URLWithString:feedURL];
 			if (safariURL != nil && ![safariURL isFileURL])
 			{
 				if (![@"feed" isEqualToString:[safariURL scheme]])
@@ -1441,7 +1441,7 @@
 			[db beginTransaction];
 			NSString * feedTitle = [arrayOfTitles objectAtIndex:index];
 			NSString * feedURL = [arrayOfURLs objectAtIndex:index];
-			NSURL * draggedURL = [NSURL URLWithString:feedURL];
+			NSURL * draggedURL = [NSURL XP_URLWithString:feedURL];
 			if (([draggedURL scheme] != nil) && [[draggedURL scheme] isEqualToString:@"feed"])
 				feedURL = [NSString stringWithFormat:@"http:%@", [draggedURL resourceSpecifier]];
 			
