--- vienna/RichXMLParser.m	2026-04-05 14:14:13.794922733 +0000
+++ vienna/RichXMLParser.m	2026-04-06 21:48:19.125339883 +0000
@@ -291,10 +291,11 @@
 	NSArray * arrayOfTags = [XMLTag parserFromData:xmlData];
 	if (arrayOfTags != nil)
 	{
-		for (XMLTag * tag in arrayOfTags)
-		{
-			NSString * tagName = [tag name];
-
+	        NSEnumerator * enum_tag = [arrayOfTags objectEnumerator];
+	        XMLTag * tag;
+	        while ((tag = [enum_tag nextObject]))
+	        {
+	                NSString * tagName = [tag name];
 			if ([tagName isEqualToString:@"rss"] || [tagName isEqualToString:@"rdf:rdf"] || [tagName isEqualToString:@"feed"])
 			{
 				success = NO;
@@ -784,14 +785,15 @@
 	// previous one. So the array is effectively newest first.
 	NSDate * itemDate = [self lastModified];
 	if (itemDate == nil)
-		itemDate = [NSDate date];
-	for (FeedItem * anItem in items)
-	{
-		if ([anItem date] == nil)
-			[anItem setDate:itemDate];
-		itemDate = [itemDate addTimeInterval:-1.0];
-	}
-	return success;
+	        itemDate = [NSDate date];
+	NSEnumerator * enum_anItem = [items objectEnumerator];
+	FeedItem * anItem;
+	while ((anItem = [enum_anItem nextObject]))
+	{
+	        if ([anItem date] == nil)
+	                [anItem setDate:itemDate];
+	        itemDate = [itemDate addTimeInterval:-1.0];
+	}	return success;
 }
 
 /* initAtomFeed
@@ -809,7 +811,7 @@
 	NSString * linkBase = [[feedTree valueOfAttribute:@"xml:base"] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
 	if (linkBase == nil)
 		linkBase = [[self link] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
-	NSURL * linkBaseURL = (linkBase != nil) ? [NSURL URLWithString:linkBase] : nil;
+	NSURL * linkBaseURL = (linkBase != nil) ? [NSURL XP_URLWithString:linkBase] : nil;
 
 	// Iterate through the atom items
 	NSString * defaultAuthor = @"";
@@ -852,7 +854,7 @@
 				{
 					if ((linkBaseURL != nil) && ![theLink hasPrefix:@"http://"])
 					{
-						NSURL * theLinkURL = [NSURL URLWithString:theLink relativeToURL:linkBaseURL];
+						NSURL * theLinkURL = [NSURL XP_URLWithString:theLink relativeToURL:linkBaseURL];
 						[self setLink:(theLinkURL != nil) ? [theLinkURL absoluteString] : theLink];
 					}
 					else
@@ -902,10 +904,10 @@
 			NSString * entryBase = [[subTree valueOfAttribute:@"xml:base"] stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
 			if (entryBase == nil)
 				entryBase = linkBase;
-			NSURL * entryBaseURL = (entryBase != nil) ? [NSURL URLWithString:entryBase] : nil;
+			NSURL * entryBaseURL = (entryBase != nil) ? [NSURL XP_URLWithString:entryBase] : nil;
 			if ((entryBaseURL != nil) && (linkBaseURL != nil) && ([entryBaseURL scheme] == nil))
 			{
-				entryBaseURL = [NSURL URLWithString:entryBase relativeToURL:linkBaseURL];
+				entryBaseURL = [NSURL XP_URLWithString:entryBase relativeToURL:linkBaseURL];
 				if (entryBaseURL != nil)
 					entryBase = [entryBaseURL absoluteString];
 			}
@@ -963,9 +965,9 @@
 						NSString * theLink = [[subItemTree valueOfAttribute:@"href"] stringByUnescapingExtendedCharacters];
 						if (theLink != nil)
 						{
-							if ((entryBaseURL != nil) && ([[NSURL URLWithString:theLink] scheme] == nil))
+							if ((entryBaseURL != nil) && ([[NSURL XP_URLWithString:theLink] scheme] == nil))
 							{
-								NSURL * theLinkURL = [NSURL URLWithString:theLink relativeToURL:entryBaseURL];
+								NSURL * theLinkURL = [NSURL XP_URLWithString:theLink relativeToURL:entryBaseURL];
 								[newItem setEnclosure:(theLinkURL != nil) ? [theLinkURL absoluteString] : theLink];
 							}
 							else
@@ -979,9 +981,9 @@
 						NSString * theLink = [[subItemTree valueOfAttribute:@"href"] stringByUnescapingExtendedCharacters];
 						if (theLink != nil)
 						{
-							if ((entryBaseURL != nil) && ([[NSURL URLWithString:theLink] scheme] == nil))
+							if ((entryBaseURL != nil) && ([[NSURL XP_URLWithString:theLink] scheme] == nil))
 							{
-								NSURL * theLinkURL = [NSURL URLWithString:theLink relativeToURL:entryBaseURL];
+								NSURL * theLinkURL = [NSURL XP_URLWithString:theLink relativeToURL:entryBaseURL];
 								[newItem setLink:(theLinkURL != nil) ? [theLinkURL absoluteString] : theLink];
 							}
 							else
