--- vienna/AppController.m
+++ vienna/AppController.m
@@ -58,7 +58,7 @@
 #import "ClickableProgressIndicator.h"
 #import "SearchPanel.h"
 #import "SearchMethod.h"
-#import <Sparkle/Sparkle.h>
+// #import <Sparkle/Sparkle.h> — stubs.h pre-included
 #import <WebKit/WebKit.h>
 #include <mach/mach_port.h>
 #include <mach/mach_interface.h>
@@ -121,6 +121,9 @@
 	-(void)searchArticlesWithString:(NSString *)searchString;
 	-(void)sourceWindowWillClose:(NSNotification *)notification;
 	-(IBAction)cancelAllRefreshesToolbar:(id)sender;
+	-(void)showSystemProfileInfoAlert;
+	-(BOOL)handleKeyDown:(unichar)keyChar withFlags:(NSUInteger)flags;
+	-(IBAction)markReadToggle:(id)sender;
 @end
 
 // Static constant strings that are typically never tweaked
@@ -160,7 +163,7 @@
 	@"EEE dd MMM yy HH:mmss",      @"EEE dd MMM yyyy HH:mmss",
 	@"EEEE dd MMMM yy",            @"EEEE dd MMMM yyyy",
 };
-static const size_t kNumberOfDateFormatters = sizeof(kDateFormats) / sizeof(kDateFormats[0]);
+#define kNumberOfDateFormatters 26
 // C array of NSDateFormatter's : creating a NSDateFormatter is very expensive, so we create
 //  those we need early in the program launch and keep them in memory.
 static NSDateFormatter * dateFormatterArray[kNumberOfDateFormatters];
@@ -168,8 +171,6 @@
 static NSLock * dateFormatters_lock;
 static NSLocale * enUSLocale;
 
-@synthesize rssFeed = _rssFeed;
-
 /* init
  * Class instance initialisation.
  */
@@ -198,14 +199,8 @@
 -(void)awakeFromNib
 {
 
-#if ( MAC_OS_X_VERSION_MAX_ALLOWED < 1070 && !defined(NSWindowCollectionBehaviorFullScreenPrimary) )
-    enum {
-        NSWindowCollectionBehaviorFullScreenPrimary = (1 << 7)
-    };
-#endif
-	
     //Enable FullScreen Support if we are on Lion 10.7.x
-    [mainWindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+    [mainWindow XP_setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
   	
 
 	Preferences * prefs = [Preferences standardPreferences];
@@ -509,8 +504,8 @@
 	[mainWindow setToolbar:toolbar];
 	
 	// Give the status bar and filter string an embossed look
-	[[statusText cell] setBackgroundStyle:NSBackgroundStyleRaised];
-	[[currentFilterTextField cell] setBackgroundStyle:NSBackgroundStyleRaised];
+	[[statusText cell] XP_setBackgroundStyle:NSBackgroundStyleRaised];
+	[[currentFilterTextField cell] XP_setBackgroundStyle:NSBackgroundStyleRaised];
 	[currentFilterTextField setStringValue:@""];
 	
 	// Preload dictionary of standard URLs
@@ -569,23 +564,6 @@
 	// Add the app to the status bar if needed.
 	[self showAppInStatusBar];
 	
-	// Growl initialization
-	NSBundle *mainBundle = [NSBundle mainBundle];
-	NSString *path = [[mainBundle privateFrameworksPath] stringByAppendingPathComponent:@"Growl.framework"];
-	LOG_NS(@"path: %@", path);
-	NSBundle *growlFramework = [NSBundle bundleWithPath:path];
-	if([growlFramework load])
-	{
-		NSDictionary *infoDictionary = [growlFramework infoDictionary];
-		LOG_NS(@"Using Growl.framework %@ (%@)",
-			  [infoDictionary objectForKey:@"CFBundleShortVersionString"],
-			  [infoDictionary objectForKey:(NSString *)kCFBundleVersionKey]);
-
-		Class GAB = NSClassFromString(@"GrowlApplicationBridge");
-		if([GAB respondsToSelector:@selector(setGrowlDelegate:)])
-			[GAB performSelector:@selector(setGrowlDelegate:) withObject:self];
-	}
-	
 	// Start the check timer
 	[self handleCheckFrequencyChange:nil];
 	
@@ -721,7 +699,7 @@
 		[activityWindow performClose:self];
 		
 		// Put back the original app icon
-		[[NSApp dockTile] setBadgeLabel:nil];
+		[NSApp XP_setBadgeLabel:nil];
 		
 		// Save the open tabs
 		[browserView saveOpenTabs];
@@ -1718,16 +1696,7 @@
  */
 -(void)growlNotify:(id)notifyContext title:(NSString *)title description:(NSString *)description notificationName:(NSString *)notificationName
 {
-	Class GAB = NSClassFromString(@"GrowlApplicationBridge");
-	if([GAB respondsToSelector:@selector(notifyWithTitle:description:notificationName:iconData:priority:isSticky:clickContext:identifier:)])
-					[GAB setShouldUseBuiltInNotifications:NO];
-					[GAB		notifyWithTitle:title
-									description:description
-							   notificationName:notificationName
-									   iconData:nil
-									   priority:0
-									   isSticky:NO
-								   clickContext:notifyContext];
+  NSLog(@"Growl Notification: %@ %@", title, description);
 }
 
 /* growlNotificationWasClicked
@@ -2017,7 +1986,7 @@
 		& MA_NewArticlesNotification_Badge) == 0)
 	{
 		// Remove the badge if there was one.
-		[[NSApp dockTile] setBadgeLabel:nil];
+		[NSApp XP_setBadgeLabel:nil];
 	}
 	else
 	{
@@ -2031,32 +2000,32 @@
  */
 -(void)showUnreadCountOnApplicationIconAndWindowTitle
 {
-	@synchronized([NSApp dockTile]) {
+	@synchronized(self) {
 	int currentCountOfUnread = [db countOfUnread];
 	if (currentCountOfUnread == lastCountOfUnread)
 		return;
 	lastCountOfUnread = currentCountOfUnread;
-	
+
 	// Always update the app status icon first
 	[self setAppStatusBarIcon];
-	
+
 	// Don't show a count if there are no unread articles
 	if (currentCountOfUnread <= 0)
 	{
-		[[NSApp dockTile] setBadgeLabel:nil];
+		[NSApp XP_setBadgeLabel:nil];
 		[mainWindow setTitle:[self appName]];
-		return;	
-	}	
-	
+		return;
+	}
+
 	[mainWindow setTitle:[NSString stringWithFormat:@"%@ (%i %@)", [self appName], currentCountOfUnread, NSLocalizedString(@"Unread", nil)]];
-	
+
 	// Exit now if we're not showing the unread count on the application icon
 	if (([[Preferences standardPreferences] newArticlesNotification]
 		& MA_NewArticlesNotification_Badge) ==0)
 			return;
-	
+
 	NSString * countdown = [NSString stringWithFormat:@"%i", currentCountOfUnread];
-	[[NSApp dockTile] setBadgeLabel:countdown];
+	[NSApp XP_setBadgeLabel:countdown];
 
 	} // @synchronized
 }
@@ -2897,10 +2866,10 @@
 	}
 	
 	// Create then select the new folder.
-	__block NSInteger folderId;
-	[db doTransactionWithBlock:^(BOOL *rollback) {
-		folderId = [db addGoogleReaderFolder:title underParent:parentId afterChild:predecessorId subscriptionURL:url];
-	}]; //end transaction block
+	NSInteger folderId;
+	[db beginTransaction];
+	folderId = [db addGoogleReaderFolder:title underParent:parentId afterChild:predecessorId subscriptionURL:url];
+	[db commitTransaction];
 	
 	
 	if (folderId != -1)
@@ -2947,10 +2916,10 @@
 	}
 	else
 	{ //creates locally
-		__block NSInteger folderId;
-		[db doTransactionWithBlock:^(BOOL *rollback) {
-			folderId = [db addRSSFolder:[Database untitledFeedFolderName] underParent:parentId afterChild:predecessorId subscriptionURL:urlString];
-		}]; //end transaction block
+		NSInteger folderId;
+		[db beginTransaction];
+		folderId = [db addRSSFolder:[Database untitledFeedFolderName] underParent:parentId afterChild:predecessorId subscriptionURL:urlString];
+		[db commitTransaction];
 
 		if (folderId != -1)
 		{
