--- vienna/AsyncConnection.h	2026-04-07 13:21:55.960303463 +0000
+++ vienna/AsyncConnection.h	2026-04-07 13:46:30.767943175 +0000
@@ -20,6 +20,7 @@
 
 #import <Cocoa/Cocoa.h>
 #import "ActivityLog.h"
+#import "AICURLConnection.h"
 
 // Possible return values
 typedef enum {
@@ -33,7 +34,7 @@
 } ConnectStatus;
 
 @interface AsyncConnection : NSObject {
-	NSURLConnection * connector;
+	AICURLConnection * connector;
 	NSDictionary * httpHeaders;
 	NSDictionary * responseHeaders;
 	NSMutableData * receivedData;
