--- vienna/AsyncConnection.m	2026-04-07 13:21:55.960804957 +0000
+++ vienna/AsyncConnection.m	2026-04-07 13:46:31.375263151 +0000
@@ -156,7 +156,9 @@
 
 	if (httpHeaders != nil)
 	{
-		for (NSString * httpFieldName in httpHeaders)
+		NSEnumerator * httpFieldNameEnum = [httpHeaders keyEnumerator];
+		NSString * httpFieldName;
+		while ((httpFieldName = [httpFieldNameEnum nextObject]))
 		{
 			NSString * fieldValue = [httpHeaders objectForKey:httpFieldName];
 			if (fieldValue != nil && ![fieldValue isBlank])
@@ -172,7 +174,7 @@
 	
 	// Changed to not use the [NSURLConnection connectionWithRequest:delegate:  and then retain,
 	// no sense putting it in an autorelease pool if it has no reason to be there
-	connector = [[NSURLConnection alloc] initWithRequest:theRequest delegate:self];
+	connector = [[AICURLConnection alloc] initWithRequest:theRequest delegate:self];
 	return connector != nil;
 }
 
@@ -242,7 +244,9 @@
 			// Add the HTTP headers details to the log for this connection for
 			// debugging purposes.
 			[headerDetail setString:NSLocalizedString(@"Headers:\n", nil)];
-			for (NSString * headerField in responseHeaders)
+			NSEnumerator * headerFieldEnum = [responseHeaders keyEnumerator];
+			NSString * headerField;
+			while ((headerField = [headerFieldEnum nextObject]))
 			{
 				[headerDetail appendFormat:@"\t%@: %@\n", headerField, [[httpResponse allHeaderFields] valueForKey:headerField]];
 			}
