--- vienna/Database.h
+++ vienna/Database.h
@@ -19,7 +19,7 @@
 //
 
 #import <Cocoa/Cocoa.h>
-#import "FMDatabase.h"
+@class FMDatabase;
 #import "Folder.h"
 #import "Field.h"
 #import "Criteria.h"
@@ -39,8 +39,6 @@
 	NSMutableDictionary * fieldsByTitle;
 	NSMutableDictionary * foldersDict;
 	NSMutableDictionary * smartfoldersDict;
-	dispatch_queue_t _transactionQueue;
-	dispatch_queue_t _execQueue;
 	Folder * trashFolder;
 	Folder * searchFolder;
 }
@@ -59,25 +57,9 @@
 -(BOOL)readOnly;
 -(void)close;
 
-/*
- * Submits a transaction block
- * An easy way to wrap things up in a transaction can be done like this:
-    [db doTransactionWithBlock:^(BOOL *rollback) {
-        ...
-        [db ...];
-        ...
-
-        if (whoopsSomethingWrongHappened) {
-            *rollback = YES;
-            return;  // leave the block
-        }
-        // etc…
-
-        ...
-
-    }];
-*/
-- (void)doTransactionWithBlock:(void (^)(BOOL *rollback))block;
+-(void)beginTransaction;
+-(void)commitTransaction;
+-(void)rollbackTransaction;
 
 // Fields functions
 -(void)addField:(NSString *)name type:(NSInteger)type tag:(NSInteger)tag sqlField:(NSString *)sqlField visible:(BOOL)visible width:(NSInteger)width;
