--- vienna/Database.h
+++ vienna/Database.h
@@ -73,6 +73,7 @@
 -(int)addFolder:(int)parentId afterChild:(int)predecessorId folderName:(NSString *)name type:(int)type canAppendIndex:(BOOL)canAppendIndex;
 -(BOOL)deleteFolder:(int)folderId;
 -(BOOL)setFolderName:(int)folderId newName:(NSString *)newName;
+-(BOOL)setFolderType:(int)folderId newType:(int)newType;
 -(BOOL)setFolderDescription:(int)folderId newDescription:(NSString *)newDescription;
 -(BOOL)setFolderHomePage:(int)folderId newHomePage:(NSString *)newLink;
 -(BOOL)setFolderFeedURL:(int)folderId newFeedURL:(NSString *)newFeedURL;
@@ -93,6 +94,7 @@
 // RSS folder functions
 +(NSString *)untitledFeedFolderName;
 -(int)addRSSFolder:(NSString *)feedName underParent:(int)parentId afterChild:(int)predecessorId subscriptionURL:(NSString *)url;
+-(int)addGoogleReaderFolder:(NSString *)feedName underParent:(int)parentId afterChild:(int)predecessorId subscriptionURL:(NSString *)url;
 
 // Search folder functions
 -(void)setSearchString:(NSString *)newSearchString;
@@ -114,4 +116,6 @@
 -(void)markArticleDeleted:(int)folderId guid:(NSString *)guid isDeleted:(BOOL)isDeleted;
 -(BOOL)isTrashEmpty;
 -(NSArray *)guidHistoryForFolderId:(int)folderId;
+-(void)markUnreadArticlesFromFolder:(Folder *)folder guidArray:(NSArray *)guidArray;
+-(void)markStarredArticlesFromFolder:(Folder *)folder guidArray:(NSArray *)guidArray;
 @end
