--- vienna/DownloadManager.m	2026-04-05 14:14:13.771512711 +0000
+++ vienna/DownloadManager.m	2026-04-06 21:48:19.937605374 +0000
@@ -286,13 +286,14 @@
 {
 	NSMutableArray * listArray = [[NSMutableArray alloc] initWithCapacity:[downloadsList count]];
 
-	for (DownloadItem * item in downloadsList)
+	NSEnumerator * enum_item = [downloadsList objectEnumerator];
+	DownloadItem * item;
+	while ((item = [enum_item nextObject]))
 		[listArray addObject:[NSArchiver archivedDataWithRootObject:item]];
 
 	[[Preferences standardPreferences] setArray:listArray forKey:MAPref_DownloadsList];
 	[listArray release];
 }
-
 /* unarchiveDownloadsList
  * Unarchive the downloads list from the preferences.
  */
@@ -301,11 +302,12 @@
 	NSArray * listArray = [[Preferences standardPreferences] arrayForKey:MAPref_DownloadsList];
 	if (listArray != nil)
 	{
-		for (NSData * dataItem in listArray)
+		NSEnumerator * enum_dataItem = [listArray objectEnumerator];
+		NSData * dataItem;
+		while ((dataItem = [enum_dataItem nextObject]))
 			[downloadsList addObject:[NSUnarchiver unarchiveObjectWithData:dataItem]];
 	}
 }
-
 /* removeItem
  * Remove the specified item from the list.
  */
@@ -334,7 +336,7 @@
  */
 -(void)downloadFile:(NSString *)filename fromURL:(NSString *)url
 {
-	NSURLRequest * theRequest = [NSURLRequest requestWithURL:[NSURL URLWithString:url] cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:60.0];
+	NSURLRequest * theRequest = [NSURLRequest requestWithURL:[NSURL XP_URLWithString:url] cachePolicy:NSURLRequestUseProtocolCachePolicy timeoutInterval:60.0];
 	NSURLDownload * theDownload = [[NSURLDownload alloc] initWithRequest:theRequest delegate:self];	
 	if (theDownload)
 	{
