--- vienna/FilterView.m
+++ vienna/FilterView.m
@@ -40,7 +40,7 @@
 	backgroundBrush = [[NSImage alloc] initWithContentsOfFile: backgroundBrushURL ];
 
 	// Give the label the typical embossed look
-	[[filterByLabel cell] setBackgroundStyle:NSBackgroundStyleRaised];
+	[[filterByLabel cell] XP_setBackgroundStyle:NSBackgroundStyleRaised];
 	// Make sure we localise the label
 	[filterByLabel setStringValue:NSLocalizedString(@"Filter by:", nil)];
 
