--- vienna/Folder.h	2026-04-07 13:21:55.991217328 +0000
+++ vienna/Folder.h	2026-04-08 12:42:27.953472628 +0000
@@ -35,11 +35,13 @@
 #define MA_RSS_Folder			4
 #define MA_Trash_Folder			5
 #define MA_Search_Folder		6
+#define MA_GoogleReader_Folder	7
 
 // Macros to simplify getting folder types
 #define FolderType(f)			([(f) type])
 #define IsSmartFolder(f)		(([(f) type]) == MA_Smart_Folder)
 #define IsRSSFolder(f)			(([(f) type]) == MA_RSS_Folder)
+#define IsGoogleReaderFolder(f)	(([(f) type]) == MA_GoogleReader_Folder)
 #define IsGroupFolder(f)		(([(f) type]) == MA_Group_Folder)
 #define IsTrashFolder(f)		(([(f) type]) == MA_Trash_Folder)
 #define IsSearchFolder(f)		(([(f) type]) == MA_Search_Folder)
