--- vienna/HelperFunctions.m
+++ vienna/HelperFunctions.m
@@ -105,13 +105,9 @@
  */ 
 NSString * percentEscape(NSString *string)
 {
-	return CFBridgingRelease(CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault, (CFStringRef)string, NULL,
-												   
-												   // RFC2368 says all URL reserved characters must be encoded
-												   // these are all the reserved characters from RFC3986
-												   CFSTR("/:?#[]@!$&'()*+,;="),	
-												   
-												   kCFStringEncodingUTF8));
+	return [(NSString *)CFURLCreateStringByAddingPercentEscapes(kCFAllocatorDefault, (CFStringRef)string, NULL,
+								CFSTR("/:?#[]@!$&'()*+,;="),
+								kCFStringEncodingUTF8) autorelease];
 }
 
 /* cleanedUpAndEscapedUrlFromString
@@ -195,7 +191,7 @@
 BOOL isAccessible(NSString * urlString)
 {
 	SCNetworkReachabilityRef   target;
-	SCNetworkReachabilityFlags flags = 0;
+	SCNetworkConnectionFlags flags = 0;
 	Boolean                   ok;
 	
 	NSURL * url = [NSURL URLWithString:urlString];
@@ -208,7 +204,7 @@
 
         if (!ok)
             return NO;
-        return (flags & kSCNetworkReachabilityFlagsReachable) && !(flags & kSCNetworkReachabilityFlagsConnectionRequired);
+        return (flags & kSCNetworkFlagsReachable) && !(flags & kSCNetworkFlagsConnectionRequired);
     }
     else
         return NO;
