--- vienna/main.m	2026-04-05 14:14:13.953421597 +0000
+++ vienna/main.m	2026-04-06 21:48:19.907724948 +0000
@@ -1,14 +1,8 @@
-//
-//  main.m
-//  Vienna
-//
-//  Created by Steve on Sun Mar 14 2004.
-//  Copyright (c) 2004-2005 Steve Palmer. All rights reserved.
-//
-
 #import <Cocoa/Cocoa.h>
+#import <PSMTabBarControl/PSMTabBarControl.h>
 
 int main(int argc, const char *argv[])
 {
-	return NSApplicationMain(argc, argv);
+    [PSMTabBarControl class];
+    return NSApplicationMain(argc, argv);
 }
