--- vienna/NewGroupFolder.m
+++ vienna/NewGroupFolder.m
@@ -56,10 +56,10 @@
 	
 	// Create the new folder in the database
 	Database * db = [Database sharedDatabase];
-	__block int newFolderId;
-	[db doTransactionWithBlock:^(BOOL *rollback) {
-		newFolderId = [db addFolder:parentId afterChild:-1 folderName:folderNameString type:MA_Group_Folder canAppendIndex:NO];
-	}]; //end transaction block
+	int newFolderId;
+	[db beginTransaction];
+	newFolderId = [db addFolder:parentId afterChild:-1 folderName:folderNameString type:MA_Group_Folder canAppendIndex:NO];
+	[db commitTransaction];
 
 	// Close the window
 	[NSApp endSheet:newGroupFolderWindow];
