--- vienna/Preferences.m
+++ vienna/Preferences.m
@@ -22,7 +22,7 @@
 #import "Constants.h"
 #import "Article.h"
 #import "SearchMethod.h"
-#import <Sparkle/Sparkle.h>
+// #import <Sparkle/Sparkle.h> — stubs.h pre-included
 
 // Initial paths
 NSString * MA_ApplicationSupportFolder = @"~/Library/Application Support/Vienna";
@@ -128,7 +128,7 @@
 			if (![fileManager fileExistsAtPath:profilePath isDirectory:&isDir])
 			{
 				NSError * error;
-				if (![fileManager createDirectoryAtPath:profilePath withIntermediateDirectories:YES attributes:NULL error:&error])
+				if (![fileManager XP_createDirectoryAtPath:profilePath withIntermediateDirectories:YES attributes:NULL error:&error])
 				{
 					NSLog(@"Cannot create profile folder %@: %@", profilePath, error);
 					[error release];
@@ -281,13 +281,13 @@
 	[defaultValues setObject:boolYes forKey:MAPref_CheckForNewArticlesOnStartup];
 	[defaultValues setObject:[NSNumber numberWithInt:1] forKey:MAPref_CachedFolderID];
 	[defaultValues setObject:MA_Field_Date forKey:MAPref_SortColumn];
-	[defaultValues setObject:[NSNumber numberWithInteger:MA_Default_Check_Frequency] forKey:MAPref_CheckFrequency];
+	[defaultValues setObject:[NSNumber XP_numberWithInteger:MA_Default_Check_Frequency] forKey:MAPref_CheckFrequency];
 	[defaultValues setObject:[NSNumber numberWithFloat:MA_Default_Read_Interval] forKey:MAPref_MarkReadInterval];
 	[defaultValues setObject:[NSNumber numberWithInt:MA_Default_RefreshThreads] forKey:MAPref_RefreshThreads];
 	[defaultValues setObject:[NSArray arrayWithObjects:nil] forKey:MAPref_ArticleListColumns];
 	[defaultValues setObject:MA_DefaultStyleName forKey:MAPref_ActiveStyleName];
 	[defaultValues setObject:[NSNumber numberWithFloat:1.0] forKey:MAPref_ActiveTextSizeMultiplier];
-	[defaultValues setObject:[NSNumber numberWithInteger:MA_Default_BackTrackQueueSize] forKey:MAPref_BacktrackQueueSize];
+	[defaultValues setObject:[NSNumber XP_numberWithInteger:MA_Default_BackTrackQueueSize] forKey:MAPref_BacktrackQueueSize];
 	[defaultValues setObject:[NSNumber numberWithInt:MA_FolderSort_ByName] forKey:MAPref_AutoSortFoldersTree];
 	[defaultValues setObject:boolYes forKey:MAPref_ShowFolderImages];
 	[defaultValues setObject:boolYes forKey:MAPref_UseJavaScript];
@@ -300,8 +300,8 @@
 	[defaultValues setObject:boolNo forKey:MAPref_NewFolderUI];
 	[defaultValues setObject:boolNo forKey:MAPref_UseMinimumFontSize];
 	[defaultValues setObject:[NSNumber numberWithInt:MA_Filter_All] forKey:MAPref_FilterMode];
-	[defaultValues setObject:[NSNumber numberWithInteger:MA_Default_MinimumFontSize] forKey:MAPref_MinimumFontSize];
-	[defaultValues setObject:[NSNumber numberWithInteger:MA_Default_AutoExpireDuration] forKey:MAPref_AutoExpireDuration];
+	[defaultValues setObject:[NSNumber XP_numberWithInteger:MA_Default_MinimumFontSize] forKey:MAPref_MinimumFontSize];
+	[defaultValues setObject:[NSNumber XP_numberWithInteger:MA_Default_AutoExpireDuration] forKey:MAPref_AutoExpireDuration];
 	[defaultValues setObject:MA_DefaultDownloadsFolder forKey:MAPref_DownloadsFolder];
 	[defaultValues setObject:defaultArticleSortDescriptors forKey:MAPref_ArticleSortDescriptors];
 	[defaultValues setObject:[NSDate distantPast] forKey:MAPref_LastRefreshDate];
@@ -313,7 +313,7 @@
 	[defaultValues setObject:boolYes forKey:MAPref_ShouldSaveFeedSource];
 	[defaultValues setObject:boolNo forKey:MAPref_ShouldSaveFeedSourceBackup];
 	[defaultValues setObject:[NSKeyedArchiver archivedDataWithRootObject:[SearchMethod searchAllArticlesMethod]] forKey:MAPref_SearchMethod];
-    [defaultValues setObject:[NSNumber numberWithInteger:MA_Default_ConcurrentDownloads] forKey:MAPref_ConcurrentDownloads];
+    [defaultValues setObject:[NSNumber XP_numberWithInteger:MA_Default_ConcurrentDownloads] forKey:MAPref_ConcurrentDownloads];
     [defaultValues setObject:boolNo forKey:MAPref_SyncGoogleReader];
     [defaultValues setObject:boolNo forKey:MAPref_GoogleNewSubscription];
     [defaultValues setObject:boolNo forKey:MAPref_AlwaysAcceptBetas];
@@ -348,7 +348,7 @@
  */
 -(void)setInteger:(NSInteger)value forKey:(NSString *)defaultName
 {
-	[userPrefs setObject:[NSNumber numberWithInteger:value] forKey:defaultName];
+	[userPrefs setObject:[NSNumber XP_numberWithInteger:value] forKey:defaultName];
 }
 
 /* setString
@@ -388,7 +388,7 @@
  */
 -(NSInteger)integerForKey:(NSString *)defaultName
 {
-	return [[userPrefs valueForKey:defaultName] integerValue];
+	return [[userPrefs valueForKey:defaultName] XP_integerValue];
 }
 
 /* stringForKey
@@ -1196,7 +1196,7 @@
 	if (![[NSFileManager defaultManager] fileExistsAtPath:feedSourcesFolder isDirectory:&isDirectory])
 	{
 		NSError * error = nil;
-		if (![[NSFileManager defaultManager] createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
+		if (![[NSFileManager defaultManager] XP_createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
 		{
 			NSLog(@"Could not create feed sources folder at path '%@'. Error: %@", feedSourcesFolder, [error localizedDescription]);
 		}
