--- vienna/Preferences.m
+++ vienna/Preferences.m
@@ -22,7 +22,7 @@
 #import "Constants.h"
 #import "Article.h"
 #import "SearchMethod.h"
-#import <Sparkle/Sparkle.h>
+// #import <Sparkle/Sparkle.h> — stubs.h pre-included
 
 // Initial paths
 NSString * MA_ApplicationSupportFolder = @"~/Library/Application Support/Vienna";
@@ -128,7 +128,7 @@
 			if (![fileManager fileExistsAtPath:profilePath isDirectory:&isDir])
 			{
 				NSError * error;
-				if (![fileManager createDirectoryAtPath:profilePath withIntermediateDirectories:YES attributes:NULL error:&error])
+				if (![fileManager XP_createDirectoryAtPath:profilePath withIntermediateDirectories:YES attributes:NULL error:&error])
 				{
 					NSLog(@"Cannot create profile folder %@: %@", profilePath, error);
 					[error release];
@@ -1196,7 +1196,7 @@
 	if (![[NSFileManager defaultManager] fileExistsAtPath:feedSourcesFolder isDirectory:&isDirectory])
 	{
 		NSError * error = nil;
-		if (![[NSFileManager defaultManager] createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
+		if (![[NSFileManager defaultManager] XP_createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
 		{
 			NSLog(@"Could not create feed sources folder at path '%@'. Error: %@", feedSourcesFolder, [error localizedDescription]);
 		}
