--- vienna/RefreshManager.m
+++ vienna/RefreshManager.m
@@ -54,7 +54,7 @@
 -(NSString *)getRedirectURL:(NSData *)data;
 - (void)syncFinishedForFolder:(Folder *)folder; 
 @property (nonatomic, retain) NSTimer * unsafe301RedirectionTimer;
-@property (atomic, copy) NSString *riskyIPAddress;
+@property (copy) NSString *riskyIPAddress;
 @end
 
 @implementation RefreshManager
@@ -90,15 +90,11 @@
 		[nc addObserver:self selector:@selector(handleChangeConcurrentDownloads:) name:@"MA_Notify_CowncurrentDownloadsChange" object:nil];
 		// be notified on system wake up after sleep
 		[[[NSWorkspace sharedWorkspace] notificationCenter] addObserver:self selector:@selector(handleDidWake:) name:@"NSWorkspaceDidWakeNotification" object:nil];
-		_queue = dispatch_queue_create("uk.co.opencommunity.vienna2.refresh", NULL);
 		hasStarted = NO;
 	}
 	return self;
 }
 
--(dispatch_queue_t)asyncQueue {
-	return _queue;
-}
 
 - (void)nqQueueDidFinishSelector:(ASIHTTPRequest *)request {
 	[[NSNotificationCenter defaultCenter] postNotificationName:@"MA_Notify_ArticleListStateChange" object:nil];
@@ -280,9 +276,7 @@
 			[self refreshFolderIconCacheForSubscriptions:[[Database sharedDatabase] arrayOfFolders:[folder itemId]]];
 		else if (IsRSSFolder(folder) || IsGoogleReaderFolder(folder))
 		{
-			dispatch_async(dispatch_get_main_queue(), ^{
-				[self refreshFavIcon:folder];
-			});
+			[self refreshFavIcon:folder];
 		}
 	}
 }
@@ -298,9 +292,9 @@
 	if ((IsRSSFolder(folder)||IsGoogleReaderFolder(folder)) && ([folder homePage] == nil || [[folder homePage] isBlank] || [folder hasCachedImage]))
 	{
 		Database *db = [Database sharedDatabase];
-		[db doTransactionWithBlock:^(BOOL *rollback) {
-			[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
-		}]; //end transaction block
+		[db beginTransaction];
+		[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
+		[db commitTransaction];
 		return;
 	}
 	
@@ -600,9 +594,9 @@
 	}
 
 	Database *db = [Database sharedDatabase];
-	[db doTransactionWithBlock:^(BOOL *rollback) {
-		[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
-	}]; //end transaction block
+	[db beginTransaction];
+	[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
+	[db commitTransaction];
 }
 
 // failure callback
@@ -613,20 +607,18 @@
 	[aItem appendDetail:[NSString stringWithFormat:@"%@ %@",NSLocalizedString(@"Error retrieving RSS Icon:", nil),[[request error] localizedDescription ]]];
 
 	Database *db = [Database sharedDatabase];
-	[db doTransactionWithBlock:^(BOOL *rollback) {
-		[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
-	}]; //end transaction block
+	[db beginTransaction];
+	[db clearFolderFlag:[folder itemId] flagToClear:MA_FFlag_CheckForImage];
+	[db commitTransaction];
 }
 
 - (void)syncFinishedForFolder:(Folder *)folder 
 {
     [self setFolderUpdatingFlag:folder flag:NO];
-    dispatch_async(dispatch_get_main_queue(), ^{
-		// Unread count may have changed
-		AppController *controller = APPCONTROLLER;
-		[controller setStatusMessage:nil persist:NO];
-		[controller showUnreadCountOnApplicationIconAndWindowTitle];
-	});
+	// Unread count may have changed
+	AppController *controller = APPCONTROLLER;
+	[controller setStatusMessage:nil persist:NO];
+	[controller showUnreadCountOnApplicationIconAndWindowTitle];
 }
 
 /* folderRefreshRedirect
@@ -648,9 +640,9 @@
 
         if ([[newURL host] isEqualToString:[[connector originalURL] host]] || [self canTrust301Redirects:connector])
         {
-			[db doTransactionWithBlock:^(BOOL *rollback) {
-				[db setFolderFeedURL:folderId newFeedURL:[newURL absoluteString]];
-			}]; //end transaction block
+			[db beginTransaction];
+			[db setFolderFeedURL:folderId newFeedURL:[newURL absoluteString]];
+			[db commitTransaction];
 
 			[connectorItem appendDetail:[NSString stringWithFormat:NSLocalizedString(@"Feed URL updated to %@", nil), [newURL absoluteString]]];
 		}
@@ -703,8 +695,6 @@
  */
 -(void)folderRefreshCompleted:(ASIHTTPRequest *)connector
 {
-	dispatch_async(_queue, ^() {
-		
 	Folder * folder = (Folder *)[[connector userInfo] objectForKey:@"folder"];
 	ActivityItem *connectorItem = [[connector userInfo] objectForKey:@"log"];
 	int responseStatusCode = [connector responseStatusCode];
@@ -734,9 +724,9 @@
 	if (responseStatusCode == 304)
 	{		
 		// No modification from last check
-		[db doTransactionWithBlock:^(BOOL *rollback) {
-			[db setFolderLastUpdate:folderId lastUpdate:[NSDate date]];
-		}];
+		[db beginTransaction];
+		[db setFolderLastUpdate:folderId lastUpdate:[NSDate date]];
+		[db commitTransaction];
 
 		[self setFolderErrorFlag:folder flag:NO];
 		[connectorItem appendDetail:NSLocalizedString(@"Got HTTP status 304 - No news from last check", nil)];
@@ -759,9 +749,9 @@
 	else if (responseStatusCode == 410)
 	{
 		// We got HTTP 410 which means the feed has been intentionally removed so unsubscribe the feed.
-		[db doTransactionWithBlock:^(BOOL *rollback) {
-			[db setFolderFlag:folderId flagToSet:MA_FFlag_Unsubscribed];
-		}];
+		[db beginTransaction];
+		[db setFolderFlag:folderId flagToSet:MA_FFlag_Unsubscribed];
+		[db commitTransaction];
 		[[NSNotificationCenter defaultCenter] postNotificationOnMainThreadWithName:@"MA_Notify_FoldersUpdated" object:[NSNumber numberWithInt:folderId]];
 	}
 	else if (responseStatusCode == 200)
@@ -786,10 +776,7 @@
 	}
 
 	[self syncFinishedForFolder:folder];
-
-	}); //block for dispatch_async on _queue
-};
-
+}
 -(void)finalizeFolderRefresh:(NSDictionary*)parameters;
 {	
 	
@@ -824,7 +811,7 @@
         
 		        
 		// Empty data feed is OK if we got HTTP 200
-		__block NSUInteger newArticlesFromFeed = 0;
+		NSUInteger newArticlesFromFeed = 0;
 		RichXMLParser * newFeed = [[RichXMLParser alloc] init];
 		if ([receivedData length] > 0)
 		{
@@ -921,7 +908,7 @@
 				if (articleDate == nil)
 				{
 					articleDate = itemAlternativeDate;
-					itemAlternativeDate = [itemAlternativeDate dateByAddingTimeInterval:-1.0];
+					itemAlternativeDate = [itemAlternativeDate XP_dateByAddingTimeInterval:-1.0];
 				}
 				
 				Article * article = [[[Article alloc] initWithGuid:articleGuid] autorelease];
@@ -952,49 +939,49 @@
 			if ([articleArray count] > 0u)
 			{
                 // Should we wrap the entire loop or just individual article updates?
-				[db doTransactionWithBlock:^(BOOL *rollback) {
-				NSArray * guidHistory = [db guidHistoryForFolderId:folderId];
-				[folder clearCache];
-				for (Article * article in articleArray)
-				{
-					if ([db createArticle:folderId article:article guidHistory:guidHistory] && ([article status] == MA_MsgStatus_New))
-						++newArticlesFromFeed;
-				}
-				}]; //end transaction block
+				[db beginTransaction];
+			NSArray * guidHistory = [db guidHistoryForFolderId:folderId];
+			[folder clearCache];
+			for (Article * article in articleArray)
+			{
+				if ([db createArticle:folderId article:article guidHistory:guidHistory] && ([article status] == MA_MsgStatus_New))
+					++newArticlesFromFeed;
+			}
+				[db commitTransaction];
 
 			}
 			
-			[db doTransactionWithBlock:^(BOOL *rollback) {
+			[db beginTransaction];
+        
+		// A notify is only needed if we added any new articles.
+		if ([[folder name] hasPrefix:[Database untitledFeedFolderName]] && ![feedTitle isBlank])
+		{
+			// If there's an existing feed with this title, make ours unique
+			// BUGBUG: This duplicates logic in database.m so consider moving it there.
+			NSString * oldFeedTitle = feedTitle;
+			NSString * newFeedTitle = feedTitle;
+			NSUInteger index = 1;
             
-			// A notify is only needed if we added any new articles.
-			if ([[folder name] hasPrefix:[Database untitledFeedFolderName]] && ![feedTitle isBlank])
-			{
-				// If there's an existing feed with this title, make ours unique
-				// BUGBUG: This duplicates logic in database.m so consider moving it there.
-				NSString * oldFeedTitle = feedTitle;
-				NSString * newFeedTitle = feedTitle;
-				NSUInteger index = 1;
-                
-				while (([db folderFromName:newFeedTitle]) != nil)
-					newFeedTitle = [NSString stringWithFormat:@"%@ (%li)", oldFeedTitle, index++];
-                
-				[connectorItem setName:newFeedTitle];
-				[db setFolderName:folderId newName:newFeedTitle];
-			}
-			if (feedDescription != nil)
-				[db setFolderDescription:folderId newDescription:feedDescription];
+			while (([db folderFromName:newFeedTitle]) != nil)
+				newFeedTitle = [NSString stringWithFormat:@"%@ (%li)", oldFeedTitle, index++];
             
-			if (feedLink!= nil)
-				[db setFolderHomePage:folderId newHomePage:feedLink];
+			[connectorItem setName:newFeedTitle];
+			[db setFolderName:folderId newName:newFeedTitle];
+		}
+		if (feedDescription != nil)
+			[db setFolderDescription:folderId newDescription:feedDescription];
+        
+		if (feedLink!= nil)
+			[db setFolderHomePage:folderId newHomePage:feedLink];
 
-			// Remember the last modified date
-			if (lastModifiedString != nil)
-				[db setFolderLastUpdateString:folderId lastUpdateString:lastModifiedString];
+		// Remember the last modified date
+		if (lastModifiedString != nil)
+			[db setFolderLastUpdateString:folderId lastUpdateString:lastModifiedString];
 
-			// Set the last update date for this folder.
-			[db setFolderLastUpdate:folderId lastUpdate:[NSDate date]];
-			
-			}]; //end transaction block
+		// Set the last update date for this folder.
+		[db setFolderLastUpdate:folderId lastUpdate:[NSDate date]];
+		
+			[db commitTransaction];
 
 			// Mark the feed as succeeded
 			[self setFolderErrorFlag:folder flag:NO];
@@ -1163,7 +1150,6 @@
 	unsafe301RedirectionTimer=nil;
 	[riskyIPAddress release];
 	riskyIPAddress=nil;
-	dispatch_release(_queue);
 	[super dealloc];
 }
 @end
