--- vienna/SearchMethod.m
+++ vienna/SearchMethod.m
@@ -20,7 +20,6 @@
 
 
 #import "SearchMethod.h"
-#import "HelperFunctions.h"
 #import "AppController.h"
 
 @implementation SearchMethod
@@ -120,7 +119,7 @@
 - (NSURL *)queryURLforSearchString:(NSString *)searchString;
 {
 	NSURL * queryURL;
-	NSString * temp = [[self searchQueryString] stringByReplacingOccurrencesOfString:@"$SearchTerm$" withString:searchString];
+	NSString * temp = [[self searchQueryString] XP_stringByReplacingOccurrencesOfString:@"$SearchTerm$" withString:searchString];
 	queryURL = cleanedUpAndEscapedUrlFromString(temp);
     return queryURL;
 }
