--- vienna/SearchFolder.m
+++ vienna/SearchFolder.m
@@ -470,18 +470,18 @@
 	// Set the criteria condition
 	[criteriaTree setCondition:[criteriaConditionPopup selectedTag]];
 	
-	[db doTransactionWithBlock:^(BOOL *rollback) {
+	[db beginTransaction];
 	if (smartFolderId == -1)
 	{
-		AppController * controller = APPCONTROLLER;
-		smartFolderId = [db addSmartFolder:folderName underParent:parentId withQuery:criteriaTree];
-		[controller selectFolder:smartFolderId];
+	AppController * controller = APPCONTROLLER;
+	smartFolderId = [db addSmartFolder:folderName underParent:parentId withQuery:criteriaTree];
+	[controller selectFolder:smartFolderId];
 	}
 	else
-    {
-		[db updateSearchFolder:smartFolderId withFolder:folderName withQuery:criteriaTree];
-    }
-    }]; //end transaction block
+{
+	[db updateSearchFolder:smartFolderId withFolder:folderName withQuery:criteriaTree];
+}
+	[db commitTransaction];
 
 	[criteriaTree release];
 	
