--- vienna/StringExtensions.m	2026-04-07 10:53:55.910594387 +0000
+++ vienna/StringExtensions.m	2026-04-07 11:10:02.989800946 +0000
@@ -42,7 +42,7 @@
 	baseURL = [baseURL stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
 	if (baseURL == nil)
 		return;
-	NSURL * imgBaseURL = [NSURL URLWithString:baseURL];
+	NSURL * imgBaseURL = [NSURL XP_URLWithString:baseURL];
 	
 	unsigned int textLength = [self length];
 	NSRange srchRange;
@@ -72,7 +72,7 @@
 				NSString * escapedSrcPath = [srcPath stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
 				if (escapedSrcPath != nil)
 				{
-					NSURL * imgURL = [NSURL URLWithString:escapedSrcPath relativeToURL:imgBaseURL];
+					NSURL * imgURL = [NSURL XP_URLWithString:escapedSrcPath relativeToURL:imgBaseURL];
 					if (imgURL != nil)
 					{
 						srcPath = [imgURL absoluteString];
@@ -592,7 +592,7 @@
  */
 -(NSString *)baseURL
 {
-	NSURL * url = [NSURL URLWithString:self];
+	NSURL * url = [NSURL XP_URLWithString:self];
 	return (url && [url host]) ? [url host] : self;
 }
 
