--- vienna/BrowserView.m	2026-04-05 14:14:13.766195795 +0000
+++ vienna/BrowserView.m	2026-04-06 21:48:19.224754809 +0000
@@ -309,7 +309,9 @@
 {
 	NSMutableArray *tabLinks = [[NSMutableArray alloc] initWithCapacity:[self countOfTabs]];
 	
-	for (NSTabViewItem * tabViewItem in [tabView tabViewItems])
+	NSEnumerator * enum_tabViewItem = [[tabView tabViewItems] objectEnumerator];
+	NSTabViewItem * tabViewItem;
+	while ((tabViewItem = [enum_tabViewItem nextObject]))
 	{
 		NSView<BaseView> * theView = [tabViewItem identifier];
 		NSString * tabLink = [theView viewLink];
