--- vienna/HelperFunctions.m	2026-04-05 14:14:13.781054010 +0000
+++ vienna/HelperFunctions.m	2026-04-06 21:48:19.922483937 +0000
@@ -17,7 +17,6 @@
 //  See the License for the specific language governing permissions and
 // 
 
-#import "HelperFunctions.h"
 #import <SystemConfiguration/SCNetworkReachability.h>
 #import <Carbon/Carbon.h>
 
@@ -47,7 +46,7 @@
  */
 NSString * getDefaultBrowser(void)
 {
-	NSURL * testURL = [NSURL URLWithString:@"http://example.net"];
+	NSURL * testURL = [NSURL XP_URLWithString:@"http://example.net"];
 	NSString * registeredAppURL = nil;
 	CFURLRef appURL = nil;
 
@@ -109,11 +108,12 @@
 	{
 		if (validExtensions)
 			arrayOfFiles = [arrayOfFiles pathsMatchingExtensions:validExtensions];
-		
-		for (NSString * fileName in arrayOfFiles)
+
+		NSEnumerator * enum_fileName = [arrayOfFiles objectEnumerator];
+		NSString * fileName;
+		while ((fileName = [enum_fileName nextObject]))
 		{
-			NSString * fullPath = [path stringByAppendingPathComponent:fileName];
-			BOOL isDirectory;
+			NSString * fullPath = [path stringByAppendingPathComponent:fileName];			BOOL isDirectory;
 			
 			if ([fileManager fileExistsAtPath:fullPath isDirectory:&isDirectory] && (isDirectory == foldersOnly))
 			{
@@ -132,7 +132,7 @@
 BOOL isAccessible(NSString * urlString)
 {
 	SCNetworkConnectionFlags flags;
-	NSURL * url = [NSURL URLWithString:urlString];
+	NSURL * url = [NSURL XP_URLWithString:urlString];
 	
 	if (!SCNetworkCheckReachabilityByName([[url host] UTF8String], &flags))
 		return NO;
@@ -238,3 +238,4 @@
 		err = AHGotoPage(myBookName, pagePath, anchorName);
 	return err;
 }
+
