--- vienna/InfoWindow.m	2026-04-05 14:14:13.782019711 +0000
+++ vienna/InfoWindow.m	2026-04-06 21:48:19.503669657 +0000
@@ -302,7 +302,7 @@
 		NSString * url = [[urlField stringValue] trim];
 		
 		// Escape any special query characters in the URL, because the URL itself will be in a query.
-		NSString * query = [[NSURL URLWithString:url] query];
+		NSString * query = [[NSURL XP_URLWithString:url] query];
 		if (query != nil)
 		{
 			NSMutableString * escapedQuery = [NSMutableString stringWithString:query];
