--- vienna/InfoWindow.m	2026-04-07 13:09:04.215438093 +0000
+++ vienna/InfoWindow.m	2026-04-07 13:11:21.940302653 +0000
@@ -318,7 +318,7 @@
 		NSString * url = [[urlField stringValue] trim];
 		
 		// Escape any special query characters in the URL, because the URL itself will be in a query.
-		NSString * query = [[NSURL URLWithString:url] query];
+		NSString * query = [[NSURL XP_URLWithString:url] query];
 		if (query != nil)
 		{
 			NSMutableString * escapedQuery = [NSMutableString stringWithString:query];
