--- vienna/KeyChain.m	2026-04-05 14:14:13.782628623 +0000
+++ vienna/KeyChain.m	2026-04-06 21:48:20.190294263 +0000
@@ -29,7 +29,7 @@
  */
 +(NSString *)getPasswordFromKeychain:(NSString *)username url:(NSString *)url
 {
-	NSURL * secureUrl = [NSURL URLWithString:url];
+	NSURL * secureUrl = [NSURL XP_URLWithString:url];
 	const char * cServiceName = [[secureUrl host] UTF8String];
 	const char * cUsername = [username UTF8String];
 	int portNumber = [secureUrl port] ? [[secureUrl port] intValue] : 80;
@@ -76,7 +76,7 @@
  */
 +(void)setPasswordInKeychain:(NSString *)password username:(NSString *)username url:(NSString *)url
 {
-	NSURL * secureUrl = [NSURL URLWithString:url];
+	NSURL * secureUrl = [NSURL XP_URLWithString:url];
 	const char * cServiceName = [[secureUrl host] UTF8String];
 	const char * cUsername = [username UTF8String];
 	const char * cPath = "";
