--- vienna/Preferences.m	2026-04-07 13:09:04.220368053 +0000
+++ vienna/Preferences.m	2026-04-07 13:11:22.302820476 +0000
@@ -22,7 +22,6 @@
 #import "Constants.h"
 #import "Message.h"
 #import "SearchMethod.h"
-#import <Sparkle/Sparkle.h>
 
 // Initial paths
 NSString * MA_ApplicationSupportFolder = @"~/Library/Application Support/Vienna";
@@ -1039,7 +1038,7 @@
 	if (![[NSFileManager defaultManager] fileExistsAtPath:feedSourcesFolder isDirectory:&isDirectory])
 	{
 		NSError * error = nil;
-		if (![[NSFileManager defaultManager] createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
+		if (![[NSFileManager defaultManager] XP_createDirectoryAtPath:feedSourcesFolder withIntermediateDirectories:YES attributes:nil error:&error])
 		{
 			NSLog(@"Could not create feed sources folder at path '%@'. Error: %@", feedSourcesFolder, [error localizedDescription]);
 		}
