--- vienna/SplitViewExtensions.m	2026-04-05 14:13:20.575135690 +0000
+++ vienna/SplitViewExtensions.m	2026-04-06 21:48:20.013063851 +0000
@@ -34,8 +34,10 @@
 {
 	NSMutableArray * viewRects = [NSMutableArray array];
 	NSRect frame;
-	
-	for (NSView * view in [self subviews])
+
+	NSEnumerator * enum_view = [[self subviews] objectEnumerator];
+	NSView * view;
+	while ((view = [enum_view nextObject]))
 	{
 		if ([self isSubviewCollapsed:view])
 			frame = NSZeroRect;
@@ -45,7 +47,6 @@
 	}
 	return viewRects;
 }
-
 /* setLayout
  * Sets the splitview layout from the specified array
  */
