--- vienna/TabbedWebView.m	2026-04-07 10:53:55.912601741 +0000
+++ vienna/TabbedWebView.m	2026-04-07 13:11:21.683859504 +0000
@@ -172,7 +172,7 @@
 		// Indicate a redirect for a feed
 		[self setIsFeedRedirect:YES];
 
-		[controller openURLInDefaultBrowser:[NSURL URLWithString:[NSString stringWithFormat:@"feed://%@", linkPath]]];
+		[controller openURLInDefaultBrowser:[NSURL XP_URLWithString:[NSString stringWithFormat:@"feed://%@", linkPath]]];
 		[listener ignore];
 		return;
 	}
