--- vienna/TabbedWebView.m	2026-04-05 14:14:13.269324291 +0000
+++ vienna/TabbedWebView.m	2026-04-06 21:48:19.233676234 +0000
@@ -169,7 +169,7 @@
 		// Indicate a redirect for a feed
 		[self setIsFeedRedirect:YES];
 
-		[controller openURLInDefaultBrowser:[NSURL URLWithString:[NSString stringWithFormat:@"feed://%@", linkPath]]];
+		[controller openURLInDefaultBrowser:[NSURL XP_URLWithString:[NSString stringWithFormat:@"feed://%@", linkPath]]];
 		[listener ignore];
 		return;
 	}
