--- vienna/TableViewExtensions.m	2026-04-05 14:14:13.882347163 +0000
+++ vienna/TableViewExtensions.m	2026-04-06 21:48:19.995226002 +0000
@@ -96,13 +96,14 @@
  */
 -(void)localiseHeaderStrings
 {
-	for (NSTableColumn * aColumn in [self tableColumns])
+	NSEnumerator * enum_aColumn = [[self tableColumns] objectEnumerator];
+	NSTableColumn * aColumn;
+	while ((aColumn = [enum_aColumn nextObject]))
 	{
 		id headerCell = [aColumn headerCell];
 		[headerCell setStringValue:NSLocalizedString([headerCell stringValue], nil)];
 	}
 }
-
 /* menuForEvent
  * Handle menu by moving the selection.
  */
