--- vienna/TreeNode.m	2026-04-07 10:53:55.914407639 +0000
+++ vienna/TreeNode.m	2026-04-07 11:10:02.786034469 +0000
@@ -154,13 +154,14 @@
 		return self;
 	
 	TreeNode * theNode;
-	
-	for (TreeNode * node in children)
+
+	NSEnumerator * enum_node = [children objectEnumerator];
+	TreeNode * node;
+	while ((node = [enum_node nextObject]))
 	{
 		if ((theNode = [node nodeFromID:n]) != nil)
 			return theNode;
-	}
-	return nil;
+	}	return nil;
 }
 
 /* childByName
@@ -168,7 +169,9 @@
  */
 -(TreeNode *)childByName:(NSString *)childName
 {
-	for (TreeNode * node in children)
+	NSEnumerator * enum_node = [children objectEnumerator];
+	TreeNode * node;
+	while ((node = [enum_node nextObject]))
 	{
 		if ([childName isEqual:[node nodeName]])
 			return node;
@@ -348,7 +351,9 @@
 	}
 
 	// Make sure the children do the same in case this is being called on a folder node.
-	for (TreeNode * node in children)
+	NSEnumerator * enum_children = [children objectEnumerator];
+	TreeNode * node;
+	while ((node = [enum_children nextObject]))
 		[node stopAndReleaseProgressIndicator];
 }	
 
