--- vienna/Preferences/AdvancedPreferencesViewController.m
+++ vienna/Preferences/AdvancedPreferencesViewController.m
@@ -23,20 +23,18 @@
 #import "Preferences.h"
 
 @interface AdvancedPreferencesViewController ()
-
+- (void)initializePreferences;
 @end
 
 @implementation AdvancedPreferencesViewController
 
 
-- (instancetype)init {
+- (id)init {
     return [super initWithNibName:@"AdvancedPreferencesView" bundle:nil];
 }
 
 - (void)viewWillAppear {
-    if([NSViewController instancesRespondToSelector:@selector(viewWillAppear)]) {
-        [super viewWillAppear];
-    }
+    [super viewWillAppear];
     // Do view setup here.
     [self initializePreferences];
 }
