--- vienna/Preferences/AdvancedPreferencesViewController.m
+++ vienna/Preferences/AdvancedPreferencesViewController.m
@@ -23,20 +23,18 @@
 #import "Preferences.h"
 
 @interface AdvancedPreferencesViewController ()
-
+- (void)initializePreferences;
 @end
 
 @implementation AdvancedPreferencesViewController
 
 
-- (instancetype)init {
+- (id)init {
     return [super initWithNibName:@"AdvancedPreferencesView" bundle:nil];
 }
 
 - (void)viewWillAppear {
-    if([NSViewController instancesRespondToSelector:@selector(viewWillAppear)]) {
-        [super viewWillAppear];
-    }
+    [super viewWillAppear];
     // Do view setup here.
     [self initializePreferences];
 }
@@ -50,7 +48,7 @@
 
 - (NSImage *)toolbarItemImage
 {
-    return [NSImage imageNamed:NSImageNameAdvanced];
+    return [NSImage imageNamed:@"advancedPrefImage"];
 }
 
 - (NSString *)toolbarItemLabel
@@ -102,6 +100,6 @@
 
 
 -(IBAction)changeConcurrentDownloads:(id)sender {
-    [[Preferences standardPreferences] setConcurrentDownloads:[[concurrentDownloads titleOfSelectedItem] integerValue]];
+    [[Preferences standardPreferences] setConcurrentDownloads:[[concurrentDownloads titleOfSelectedItem] XP_integerValue]];
 }
 @end
