--- vienna/Preferences/SyncingPreferencesViewController.m
+++ vienna/Preferences/SyncingPreferencesViewController.m
@@ -25,6 +25,8 @@
 #import "SyncingPreferencesViewController.h"
 
 @interface SyncingPreferencesViewController ()
+-(void)handleServerTextDidChange:(NSNotification *)aNotification;
+-(void)handleGoogleAuthFailed:(NSNotification *)nc;
 
 @end
 
@@ -34,14 +36,12 @@
 @synthesize syncButton;
 
 
-- (instancetype)init {
+- (id)init {
     return [super initWithNibName:@"SyncingPreferencesView" bundle:nil];
 }
 
 - (void)viewWillAppear {
-    if([NSViewController instancesRespondToSelector:@selector(viewWillAppear)]) {
-        [super viewWillAppear];
-    }
+    [super viewWillAppear];
     // Do view setup here.
     sourcesDict = nil;
     // Implement this method to handle any initialization after your window controller's window has been loaded from its nib file.
